app/src/main/java/ch/epfl/sdp/drone3d/service/impl/drone/DroneExecutorImpl.kt

```diff
@@ -25,135 +26,175 @@ import kotlin.math.roundToInt
+        val completable =
+            getConnectedInstance() ?: throw IllegalStateException("Could not query drone instance")
+        val instance =
+            service.provideDrone() ?: throw IllegalStateException("Could not query drone instance")
 
         data.addSubscription(
-                completable
-                        .andThen(instance.mission.startMission())
-                        .subscribe(
-                                {
-                                    data.getMutableMissionPaused().postValue(false)
-                                    ToastHandler.showToastAsync(context, R.string.drone_mission_success)
-                                },
-                                {
-                                    ToastHandler.showToastAsync(context, R.string.drone_mission_error)
-                                }
-                        )
+            completable
+                .andThen(instance.mission.startMission())
+                .subscribe(
+                    {
+                        data.getMutableMissionPaused().postValue(false)
+                        ToastHandler.showToastAsync(context, R.string.drone_mission_success)
+                    },
+                    {
+                        ToastHandler.showToastAsync(context, R.string.drone_mission_error)
+                    }
+                )
         )
     }
 
     override fun returnToHomeLocationAndLand(context: Context) {
         val returnLocation = data.getHomeLocation().value
-                ?: throw IllegalStateException(context.getString(R.string.drone_home_error))
+            ?: throw IllegalStateException(context.getString(R.string.drone_home_error))
 
-        val completable = getConnectedInstance() ?: throw IllegalStateException("Could not query drone instance")
-        val instance = service.provideDrone() ?: throw IllegalStateException("Could not query drone instance")
+        val completable =
+            getConnectedInstance() ?: throw IllegalStateException("Could not query drone instance")
+        val instance =
+            service.provideDrone() ?: throw IllegalStateException("Could not query drone instance")
 
         goToLocation(context, completable, instance, returnLocation)
     }
 
     override fun returnToUserLocationAndLand(context: Context) {
-        // TODO Query user position
-        val returnLocation = Telemetry.Position(.0, .0, 10f, 10f)
-
-        val completable = getConnectedInstance() ?: throw IllegalStateException("Could not query drone instance")
-        val instance = service.provideDrone() ?: throw IllegalStateException("Could not query drone instance")
+        if (!locationService.isLocationEnabled()) {
+            // What TODO when cant retrieve user location ?
+            return
+        }
+        val userPosition = locationService.getCurrentLocation()!!
+        val returnLocation =
+            Telemetry.Position(userPosition.latitude, userPosition.longitude, 10f, 10f)
```

Comment:
I think the relative altitude should be 0 in the telemetry position and the absolute altitude optional if possible.